activitysim/core/workflow/state.py

```diff
@@ -714,7 +714,18 @@ def get_pyarrow(
         if t is None:
             raise KeyError(tablename)
         if isinstance(t, pd.DataFrame):
-            t = pa.Table.from_pandas(t, preserve_index=True, columns=columns)
+            df = t
+            try:
+                t = pa.Table.from_pandas(df, preserve_index=True, columns=columns)
+            except (pa.ArrowTypeError, pa.ArrowInvalid):
+                # if there are object columns, try to convert them to categories
+                df = df.copy()
```

Comment:
I saw your latest comment about significantly longer run time with this PR. I noticed you are calling copy() here. In pandas 2.0 [copy() defaults to a deep copy](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.copy.html#pandas.DataFrame.copy). I wonder if this contributed to the run time?

Reply:
I don't think this is causing the problem.  This code only executes in the `write_tables` step at the end of the model run.